tests/test_grammar_formal.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import absolute_import
+
+import os
+from unittest import TestCase, main
+
+from lark import lark, Lark, UnexpectedToken
+from lark.load_grammar import GrammarError
+
+
+# Based on TestGrammar, with lots of tests that can't be run elided.
+class TestGrammarFormal(TestCase):
+    def setUp(self):
+        lark_path = os.path.join(os.path.dirname(lark.__file__), 'grammars/lark.lark')
+        # lark_path = os.path.join(os.path.dirname(lark.__file__), 'grammars/lark.lark-ORIG')
+        with open(lark_path, 'r') as f:
+            self.lark_grammar = "\n".join(f.readlines())
```

Comment:
Why not just f.read() ?

Reply:
A bad idiom that I use all over the place :-)  I'll fix it up.

Reply:
Fixed in 9493f81e9eab6dbe42096938bbcfc1f0cf5a2135.

Reply:
Very bad idiom :)